:book:[Docs] Update quickstart documentation (#1225)
operator-framework/operator-controller: README.md

curl -L -s https://github.com/operator-framework/operator-controller/releases/latest/download/install.sh | bash -s
```

## Getting Started with OLM v1

> The following steps will guide you through the process of catalog deployment, and extension installation, upgrade (or downgrade), and uninstallation.

Reviewer: Nit: I find that turning sentences with a few embedded clauses into a list makes it a little easier to scan.
```suggestion
The following steps will guide you through the following processes:

 * Deploying a catalog
 * Installing, upgrading, or downgrading an extension
 * Deleting catalogs and extensions
```

EOF
```

Once the catalog is unpacked successfully, its content will be available for installation.

```bash
# Wait for the ClusterCatalog to be unpacked
kubectl wait --for=condition=Unpacked=True clustercatalog/operatorhubio --timeout=60s
```

### Install a Cluster Extension

For simplicity, the following example manifest includes all necessary resources to install the ArgoCD operator. 
The manifest includes installation namespace, installer service account and associated minimal set of RBAC permissions 
needed for installation, and the ClusterExtension resource, which specifies the name and version of the extension to install.
More information on installing extensions can be found [here](./docs/Tasks/installing-an-extension.md).

```bash
# Apply the sample ClusterExtension. Manifest already includes
# namespace and adequately privileged service account

> kubectl apply -f config/samples/olm_v1alpha1_clusterextension.yaml

Reviewer 2: should we make this a direct link so that users can use this file without having to clone the operator-controller repo first?

Author: I'm an idiot - I focused on the green part rather than the line. My bad.

```

> ### Upgrade/Downgrade the Cluster Extension

Reviewer 2: For downgrades I think there are some specific overrides a user would need to put in place, namely:
- Ignoring the catalog provided upgrade constraints
- Disabling the CRD checks

Reviewer 2: Maybe it is worth splitting this into distinct "Upgrade" and "Downgrade" steps to highlight that, while possible, downgrades are not allowed by default and require intentional overrides from the user.

Author: I'll factor it out and add a ticket for downgrades

Reviewer: Do we really need downgrades in the quickstart? I think a simple CR would be enough and then we can link to the other docs for more complicated tasks like using version ranges and downgrades. :shrug:

Author: I've remove it and added a link to downgrades (placeholder) in the upgrading task document.
I think I'll add a note on the readme just as a call out that its possible, but not really go into it.

Author: ![whynotboth](https://github.com/user-attachments/assets/aaf7bac1-560e-4508-b4e5-6431a299de41)
